pkg/controllers/metrics/state/suite_test.go

```diff
@@ -65,7 +66,7 @@ var _ = BeforeSuite(func() {
 	env = test.NewEnvironment(scheme.Scheme, apis.CRDs...)
 
 	ctx = settings.ToContext(ctx, test.Settings())
-	cloudProvider = &fake.CloudProvider{InstanceTypes: fake.InstanceTypesAssorted()}
+	cloudProvider = &fake.CloudProvider{InstanceTypes: fake.InstanceTypesAssorted(), AllowedCreateCalls: math.MaxInt}
```

Comment:
Can this be set automatically by a call to something like `NewCloudProvider`? It seems a huge pain for someone to have to know that they need to set this when they create a new fake cloudprovider

Reply:
Yeah sure, will do.